### ISSUE TITLE
Question: Can it work with HtmlWebpackPlugin hash option?

### ISSUE THREAD

**Issue by bennadel** [NekR/offline-plugin#290]
Right now, I am using the `hash` option to cache-bust my build files since I am hosting my site on GitHub pages and the build files go into the actual git repository. However, it looks like the `hash` is not being added to the asset locations as they are loaded into the SW cache. I was wondering if this SW approach is incompatible with the `hash` option?

My current webpack config:

```js
module.exports = {
	entry: {
		polyfill: "./app/main.polyfill.ts",
		vendor: "./app/main.vendor.ts",
		main: "./app/main.ts"
	},
	output: {
		filename: "[name].js",
		path: path.join( __dirname, "build" )
	},
	resolve: {
		extensions: [ ".ts", ".js" ]
	},
	module: {
		/* ... ts-loader stuff ... */
	},
	plugins: [
		new webpack.optimize.CommonsChunkPlugin({
			names: [ 
				"main",
				"vendor",
				"polyfill",
				"manifest"
			]
		}),
		new HtmlWebpackPlugin({
			filename: "../index.htm",
			template: "./app/main.htm",

			// This will append a unique query-string hash (for cache busting) to the 
			// injected files after each build. All files get the same hash, which makes
			// this DIFFERENT from using the "chunkhash" in the "output" config.
			hash: true
		}),
		new OfflinePlugin({
			relativePaths: false,
			publicPath: 'build/',
			ServiceWorker: {
				output: "../sw.js", // Relative to root output path (which is the "build" folder).
				cacheName: "incident-commander"
			},
			AppCache: false
		})
	]
};
```

Thanks!

**Comment by NekR** [comment]
Yes, `offline-plugin` assumes that no one is going to pass query params to webpack generated files because they are static. What's wrong with having unique names per build?

**Comment by bennadel** [comment]
Ah, ok thank you - that makes sense then.

I'm quite new to Webpack, so using the `hash` option seemed like the easiest way to provide cache-busting when the compiled files were being saved to repo. Otherwise, every time I hit `CMD-S` in my project, it creates an entirely new set of files in my repo (and I'd have to manually delete the old ones with the old version). Perhaps there's just a better way that I'm not familiar with.

That said, now that I am trying to use a Service Worker, perhaps removing the hash will be fine.

**Comment by bennadel** [comment]
Though, now that I say that out-loud, I think there is a plugin I can use that will clear my `build` folder on each build.

**Comment by NekR** [comment]
Yes, there are such plugins. But your use cases sounds reasonable anyway. I agree that in your case it's better/easier to have files with constant name.

You do not seem to use AppCache, but I want to point out that having dynamic query for static assets isn't compatible with AppCache, that's a partial reason why it isn't implemented in `offline-plugin`'s ServiceWorker code too.

**Comment by bennadel** [comment]
Makes total sense. I just added `rm -r ./build/*.js` to the front of my `npm run build` and that clears it out sufficiently for my use-case. Am now seeing that the files are being successfully served from the Service Worker :D 

As an _aside_, I did have a lot of trouble getting this all to work with a `./build` folder. I finally got it working by turning off the relative-path stuff (based on another Issue that discussed an AppCache folder). Though, to be honest, I am not sure _why_ that fixed it. 

Anyway, thanks for your feedback!
